compiler/rustc_typeck/src/check/method/probe.rs

```diff
@@ -1469,7 +1470,7 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
         let cause = traits::ObligationCause::misc(self.span, self.body_id);
         let predicate = ty::Binder::dummy(trait_ref).to_poly_trait_predicate();
         let obligation = traits::Obligation::new(cause, self.param_env, predicate);
-        traits::SelectionContext::new(self).select(&obligation)
+        traits::SelectionContext::with_suggestion(self, self.is_suggestion.0).select(&obligation)
```

Comment:
Wouldn't modifying the selection context before `select`ing to be `tainted_by_errors` also work instead of adding a new field?

Reply:
Yea so looking back at it I'm not sure I even need this new field I think I actually just needed to move where I set that we were tainted by errors. I'll see if I can change it this evening so we don't need the new selection context field.

Reply:
I think what's happen is I wanted a way to distinguish inside the selection context itself whether we were creating a suggestion but forgetting that it derefs down into the function context that spawned it which already has the `tainted_by_errors` flag

Reply:
Had a look at it this evening - it starts ICEing again if I rely on `is_tainted_by_errors()` check in the overflow check I expect that the function context is set as `tainted_by_errors` elsewhere in the typechecking stage before reporting fulfilment errors so it reports an incorrect overflow error. 

I need to confirm this first however so will do some more digging.

Reply:
That sounds like a reasonable avenue.

Reply:
I can put it back however in my other implementation of this I don't rely on it anymore instead preferring to use a flag in the inferctx, I've linked my other implementation in the PR at the bottom which I think is more robust

Reply:
^ this was meant to be in reply to the other comment - will push some different changes tomorrow to review